pkg/addons/vpc_controller.go

```diff
@@ -134,15 +136,53 @@ func (v *VPCController) generateCert() error {
 		return errors.Wrap(err, "updating approval")
 	}
 
-	approvedCSR, err := csrClientSet.Get(certificateSigningRequest.Name, metav1.GetOptions{})
+	logger.Info("waiting for certificate to be available")
+
+	watchTimeout := 45 * time.Second
```

Comment:
I would make this timeout a constant at the beginning of the file with a name so we can search for it (something like `certificateWaitTimeout`). There is also the `--timeout` flag, should we use that instead?

Reply:
I agree with making it a package-level variable but using the `--timeout` option for this may not be very intuitive. Also, the watch is an internal detail and IMO the timeout shouldn't be configurable.